TidepoolKit/Model/TBloodGlucose.swift

```diff
@@ -37,4 +37,35 @@ public struct TBloodGlucose {
             self.high = high
         }
     }
+
+    public struct StartTarget: Codable, Equatable {
+        public var start: Int?
```

Comment:
I'm also a bit confused by this.  If `start` is non-optional in the constructors, why is it optional here?  Is it valid to be nil? (not blocking the review, just asking your consideration of it)

Reply:
This is another one of those tradeoffs. These data types will be used to both write data (Loop, others) and read data (Loop Follow, others) from the Tidepool backend. 

If a client is using this library to write data, then it will end up using the initializers, so it requires the `start` property. It is absolutely required by current backend validations and will fail if it is missing.

However, this will also be used to read any and all data from the Tidepool backend. Unfortunately, not all data returned by the Tidepool backend is entirely clean. That is, I know for certain that there is data in the database that does not conform to all of the current validations. (Yes, there is a tech-debt ticket to go through and "cleanup/validate" all existing data, but it is still outstanding and is not a high priority compared to getting Loop running.)

So, to be lenient with this non-conformant data, you'll notice that almost every property is optional (except for the barest minimum to allow the API to function; specifically type, subType/deliveryType, and time).

While not ideal, it is the best possible solution in my mind without: 1) trying to duplicate all validations from the backend, and 2) failing every single slightest validation error on legacy data from the backend.

Reply:
I buy that.